service/matching/task_queue_manager.go

```diff
@@ -827,9 +847,20 @@ func (c *taskQueueManagerImpl) fetchUserData(ctx context.Context) error {
 			TaskQueue:                fetchSource,
 			TaskQueueType:            enumspb.TASK_QUEUE_TYPE_WORKFLOW,
 			LastKnownUserDataVersion: knownUserData.GetVersion(),
-			WaitNewData:              !firstCall,
+			WaitNewData:              hasFetchedUserData,
 		})
 		if err != nil {
+			switch err.(type) {
```

Comment:
I think we should use `errors.As` for these branches in case someone wraps the errors

Reply:
I don't think anything wraps these but okay